### PR TITLE
[Issue-59]: Refresh token

### PR DESCRIPTION
**Pull request by ltuffery**
# Description
Add refresh token

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added (if applicable).
- [x] The code formatted (e.g. with `npm run format`)

### CODE REVIEW

File: api/src/routes.php [hunk 1]
```diff
@@ -23,6 +24,7 @@
 Flight::group('/auth', function () {
     Flight::route('POST /register', [RegisterController::class, 'store']);
     Flight::route('POST /login', [AuthenticatedSessionController::class, 'store']);
+    Flight::route('POST /token', [RefreshTokenController::class, 'store']);
```
Comment (ltuffery) [inline api/src/routes.php:27]:
```suggestion
    Flight::route('POST /refresh', [RefreshTokenController::class, 'store']);
```

File: frontend/src/services/auth.js [hunk 2]
```diff
@@ -27,16 +27,27 @@ export const getToken = () => {
...
+
+    const res = await Api.post('/auth/token').send({ refresh: localStorage.getItem('refresh') })
+
+    if (res.status == 401) return false
+
+    const data = await res.json()
+
+    localStorage.setItem('jwt', data.token)
+
+    return true
```
Comment (ltuffery) [inline frontend/src/services/auth.js:50]: Create a function to check if the session update is possible, and if so, update it

File: api/src/Controllers/AuthenticatedSessionController.php [hunk 3]
```diff
@@ -39,6 +39,7 @@ public function store(): void
             Flight::json([
                 'success' => true,
                 'token' => $user->generateJWT(),
+                'refresh' => $user->generateRefreshJWT($request->ip),
```
Comment (ltuffery) [inline api/src/Controllers/AuthenticatedSessionController.php:42]: test

**Review by swotex — APPROVED** [review]
LGFM

**Pull request merged**